### PR TITLE
Finish implement TODO page

### LINKED ISSUE #4: to-do list
MUST HAVE
app branching off homescreen to have a to-do list

### PR DESCRIPTION
**Pull request by R745940**
Still needs adjustification for UI design

### CODE REVIEW

File: src/application/scene/Todo.java [hunk 1]
```diff
@@ -1,14 +1,186 @@
...
+    private ChoiceBox<String> choiceBox_status;
+
+
+    public void initialize() {
+
+        textField_eventName.setPromptText("TODO Events");
+
+        datePicker.setValue(LocalDate.now());
+
+        //Add TODO DOING DONE status to the choiceBox dropdown
```
Comment (bjur781) [inline src/application/scene/Todo.java]: I think the code here is self explanatory so the comment could be removed

File: src/application/scene/Todo.java [hunk 2]
```diff
@@ -1,14 +1,186 @@
...
+                case "DONE":
+                    addToList(listView_DONE, "TODO", eventMessage);
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
+
+
```
Comment (bjur781) [inline src/application/scene/Todo.java:75]: On these two move functions, i was getting lots of errors in my stacktrace when I tried to move an empty list. It didn't affect the frontend at all which is good, but still for best practice could an error catch be implemented?

File: src/application/scene/Todo.java [hunk 3]
```diff
@@ -1,14 +1,186 @@
...
+        Alert alert_clearDONEConfirmation = new Alert(Alert.AlertType.CONFIRMATION, "Do you really want to remove all Events in this area", ButtonType.YES, ButtonType.NO);
+        ButtonType result = alert_clearDONEConfirmation.showAndWait().orElse(ButtonType.NO);
+
+        if (ButtonType.YES.equals(result)) {
+            listView_DONE.getItems().clear();
+        }
+
+        try {
+            File file = new File("./src/application/scene/ToDoLists/DONE.txt");
+            FileWriter clearWriter = new FileWriter(file); //the true will append the new data
```
Comment (bjur781) [inline src/application/scene/Todo.java]: This comment is confusing. Could it be rewritten to be more clear (or removed entirely)

File: src/application/scene/css/Todo.css [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    -fx-background-color: #fff176;
+}
+
+.listView_DONE .list-cell:odd {
+    -fx-background-color: #b2fab4;
+}
+
+.listView_DONE .list-cell:even {
+    -fx-background-color: #81c784;
+}
```
Comment (bjur781) [inline src/application/scene/css/Todo.css:107]: I find the colours a bit overwhelming here. Given that yellow and green aren't found anywhere else in the app, could they be changed to something more blue/purple toned? I also think the contrast between the odd and even rows is a bit excessive, could it be toned down a little?

File: src/application/scene/fxml/Todo.fxml [hunk 5]
```diff
@@ -1,11 +1,91 @@
...
+                            <children>
+                                <ListView fx:id="listView_TODO" layoutX="58.0" layoutY="66.0" prefHeight="438.0" prefWidth="317.0" style="" styleClass="listView_TODO" AnchorPane.bottomAnchor="25.0" AnchorPane.leftAnchor="55.0" AnchorPane.rightAnchor="845.0" AnchorPane.topAnchor="100.0" />
+                                <ListView fx:id="listView_DOING" layoutX="736.0" layoutY="100.0" prefHeight="438.0" prefWidth="317.0" styleClass="listView_DOING" AnchorPane.bottomAnchor="25.0" AnchorPane.leftAnchor="450.0" AnchorPane.rightAnchor="450.0" AnchorPane.topAnchor="100.0" />
+                                <ListView fx:id="listView_DONE" layoutX="858.0" layoutY="48.0" prefHeight="438.0" prefWidth="317.0" styleClass="listView_DONE" AnchorPane.bottomAnchor="25.0" AnchorPane.leftAnchor="845.0" AnchorPane.rightAnchor="55.0" AnchorPane.topAnchor="100.0" />
+                                <Text fill="WHITE" layoutX="164.0" layoutY="90.0" strokeType="OUTSIDE" strokeWidth="0.0" styleClass="text_listTitle" text="TODO" wrappingWidth="96.75628662109375" />
```
Comment (bjur781) [inline src/application/scene/fxml/Todo.fxml]: Could these heading texts be changed to lowercase to better fit the rest of the app

File: src/application/scene/fxml/Todo.fxml [hunk 6]
```diff
@@ -1,11 +1,91 @@
...
+                        <AnchorPane layoutX="50.0" layoutY="14.0" minHeight="-Infinity" prefHeight="94.0" prefWidth="1210.0" style="-fx-background-color: #5d576e;" AnchorPane.leftAnchor="50.0" AnchorPane.rightAnchor="50.0" AnchorPane.topAnchor="14.0">
+                            <children>
+                                <TextField fx:id="textField_eventName" layoutX="47.0" layoutY="28.0" prefHeight="38.0" prefWidth="592.0" style="-fx-background-color: #e0e0e0;" styleClass="textField_eventName" />
+                                <DatePicker fx:id="datePicker" layoutX="672.0" layoutY="28.0" prefHeight="38.0" prefWidth="238.0" />
+                                <ChoiceBox fx:id="choiceBox_status" layoutX="933.0" layoutY="28.0" prefHeight="38.0" prefWidth="116.0" />
+                                <Button fx:id="button_addEvent" layoutX="1073.0" layoutY="26.0" mnemonicParsing="false" onAction="#addEvent" prefHeight="41.0" prefWidth="118.0" styleClass="button_addEvent" text="   Add" textFill="WHITE" />
+                                <ImageView fitHeight="30.0" fitWidth="30.0" layoutX="1083.0" layoutY="32.0" pickOnBounds="true" preserveRatio="true">
```
Comment (bjur781) [inline src/application/scene/fxml/Todo.fxml]: The button is a little hard to spot. Could it be changed to purple so its more clear as a call to action?

**Review by bjur781 — CHANGES_REQUESTED** [review]
Not sure how hard this would be to implement, but if you have time it'd also be nice to have the functionality to delete a single task, because right now you can only remove tasks by getting them all the way to the 'done' box and then clearing the whole box.
Great work!
edit: could you also make the whole background white (or any consistent colour) so that the whole background is the same no matter what size screen you're on? Because right now if you're on a larger screen there's the main bit in white and then grey all around it.

File: src/application/scene/Todo.java (hunk 1 above)
Comment (ghost) [reply src/application/scene/Todo.java]: Checked

File: src/application/scene/Todo.java (hunk 2 above)
Comment (ghost) [reply src/application/scene/Todo.java:75]: Patched, add exception handler

File: src/application/scene/Todo.java (hunk 3 above)
Comment (ghost) [reply src/application/scene/Todo.java]: Removed

File: src/application/scene/css/Todo.css (hunk 4 above)
Comment (ghost) [reply src/application/scene/css/Todo.css:107]: Cuz I think that kinds of colour can indicate the status of their tasks, but you are right its too bright for the whole app

File: src/application/scene/fxml/Todo.fxml (hunk 5 above)
Comment (ghost) [reply src/application/scene/fxml/Todo.fxml]: Fixed

File: src/application/scene/fxml/Todo.fxml (hunk 6 above)
Comment (ghost) [reply src/application/scene/fxml/Todo.fxml]: Fixed

**Comment by ghost** [comment]
Make changes based on preivous review
- Edit comments
- Add sidebar
- Change colour scheme

**Pull request merged**